test/suite/test_rollback_to_stable01.py

```diff
@@ -30,62 +30,78 @@
-    def large_updates(self, uri, value, ds, nrows, prepare, commit_ts):
+    def retry_rollback(self, name, txn_session, code):
+        retry_limit = 100
+        retries = 0
+        completed = False
+        saved_exception = None
+        while not completed and retries < retry_limit:
+            if retries != 0:
+                self.pr("Retrying operation for " + name)
+                if txn_session:
+                    txn_session.rollback_transaction()
+                sleep(0.1)
+                if txn_session:
+                    txn_session.begin_transaction('isolation=snapshot')
+                    self.pr("Began new transaction for " + name)
+            try:
+                code()
+                completed = True
+            except WiredTigerError as e:
+                rollback_str = wiredtiger_strerror(WT_ROLLBACK)
+                if rollback_str not in str(e):
+                    raise(e)
+                retries += 1
+                saved_exception = e
+        if not completed and saved_exception:
+            raise(saved_exception)
+
+    def large_updates(self, uri, value, ds, nrows, prepare, commit_ts, rollback_if_fail = False):
         # Update a large number of records.
         session = self.session
-        cursor = session.open_cursor(uri)
-        for i in range(1, nrows + 1):
-            session.begin_transaction()
-            cursor[ds.key(i)] = value
-            if commit_ts == 0:
-                session.commit_transaction()
-            elif prepare:
-                session.prepare_transaction('prepare_timestamp=' + timestamp_str(commit_ts-1))
-                session.timestamp_transaction('commit_timestamp=' + timestamp_str(commit_ts))
-                session.timestamp_transaction('durable_timestamp=' + timestamp_str(commit_ts+1))
-                session.commit_transaction()
-            else:
-                session.commit_transaction('commit_timestamp=' + timestamp_str(commit_ts))
-        cursor.close()
-
-    def large_modifies(self, uri, value, ds, location, nbytes, nrows, prepare, commit_ts):
+        try:
+            cursor = session.open_cursor(uri)
+            for i in range(1, nrows + 1):
+                session.begin_transaction()
+                cursor[ds.key(i)] = value
+                if commit_ts == 0:
+                    session.commit_transaction()
+                elif prepare:
+                    session.prepare_transaction('prepare_timestamp=' + timestamp_str(commit_ts-1))
+                    session.timestamp_transaction('commit_timestamp=' + timestamp_str(commit_ts))
+                    session.timestamp_transaction('durable_timestamp=' + timestamp_str(commit_ts+1))
+                    session.commit_transaction()
+                else:
+                    session.commit_transaction('commit_timestamp=' + timestamp_str(commit_ts))
+            cursor.close()
+        except WiredTigerError as e:
+            if (rollback_if_fail):
```

Comment:
I am thinking like instead of rolling back based on the user request, let it always roll back the transaction when the error is WT_ROLLBACK, as this function itself is starting the transaction. It should make the change more cleaner.

Reply:
Just didn't want to change the logic of other tests that call these methods. Do you think it's safe to always rollback?

Reply:
It should be fine. As we are failing the test anyway by raising the error in other tests where they didn't expect the rollback, so calling rollback shouldn't be a problem.

Reply:
Done, thanks!